multidimensional_sampling/README.md

```diff
@@ -0,0 +1,259 @@
+# Multidimensional sampling using scisample
+
+This repo contains a pgen script for maestro that implements several
+multidimensional sampling methods which are driven by a simple yaml
+specification included in the env/variables section of any maestro
+specification. This yaml specification is a nested list named
+`SAMPLE_DICTIONARY`, and this repo contains several examples. 
+
+It requires the [`scisample` package](https://github.com/LLNL/scisample) 
+to be installed. Some sampling methods require additional packages to
+be installed. The `best_candidate` method requires `pandas`, `numpy`, and
+`scipy`. 
+
+`scisample` can be installed with `pip install scisample`.
+
+`maestro` using `scisample` is executed as follows:
+ 
+```
+maestro run [-y] --pgen ./pgen_sample.py {sample_list.yaml},
+```
+
+where `[-y]` is an optional `auto-run` option, and `sample_list.yaml` is a maestro specification containing a `SAMPLE_DICTIONARY`.  
+
+`pgen_sample.py` currently supports four sampling modes:
```

Comment:
@crkrenn -- I just tried to test this and this file doesn't exist and needs to be added.

Reply:
Sorry Frank, the `pgen_sample.py` file is installed as part of scisample. I've updated the `README` to make this more clear. 

Can you try 'maestro run -y --pgen `which pgen_scisample.py` sample_list.yaml' from within the `multidimensional_sampling` directory with the proper venv? 

I will also try `` instead of ```.

Reply:
Not a problem -- I'll retest. :)

Reply:
@crkrenn -- This broke in my environmnet. I use `pyenv` and the `which pgen_scisample.py` fails because it resolves to a path that is a bash file (it's how `pyenv` manages the python versions). I think it would be worth having a pgen file that loads and uses scisample in a manner that you would expect a user to use it. That allows for an explicit python pgen file to be checked in here.

Reply:
I would prefer not to publish and maintain a separate copy of `pgen_scisample.py` under the "DRY" principle. 

Using `maestro` requires installation and path modification. 

Why is installing and using `scisample` different than installing and using `maestro`?

Very respectfully, 

-Chris

Reply:
So, the issue here is that pgen is not meant to be an entry point. Certain tools that manage python versions get in the way. In this case -- I use `pyenv` to manage my versions. So when I look for `which pgen_scisample.py` I get

```
which pgen_scisample.py
/Users/frank/.pyenv/shims/pgen_scisample.py
```

On Unix systems, you may be able to get past this with `realpath` but MacOSX (and most certainly Windows) will not have that. I understand that the primary use case is Unix, but these shims will get in the way if someone has some advanced Python setup. It's just something I noticed that could potentially cause issues for users.

Reply:
I understand. I've added "pgen_scisample.py" to the "multidimensional_sampling" directory, and I've updated the documentation accordingly. 

Please let me know if you have any questions or troubles. 

PS. I will be adding uqpipeline sampling support as soon as I write a few tests.

Reply:
@theciviccenter -- Awesome, thanks; I think it also provides a good example of how to use scisample that people can look at as well.

For the pipeline related stuff, do you intend to add it to this PR? Or do you want to dedicate a different PR to that?

Reply:
Please accept this pull request. I'd like to be able to point to it at my workflow talk next Monday. 

I need to update scisample to support uqpipeline before I can update this example. Thanks! -Chris

Reply:
Will do! -- thanks for being receptive to the feedback.